webapp/app/elster_client/elster_client.py

```diff
@@ -197,7 +197,7 @@ def _generate_est_request_data(form_data, year=2020):
         if isinstance(adapted_form_data[key], str):
             adapted_form_data[key] = datetime.strptime(adapted_form_data[key], '%Y-%m-%d').date()
 
-    if not adapted_form_data.get('steuernummer_exists') and adapted_form_data.get('request_new_tax_number'):
+    if adapted_form_data.get('steuernummer_exists') == 'no' and adapted_form_data.get('request_new_tax_number'):
```

Comment:
> How would we transform it to a boolean? It is a string. And if you check bool('no') you get exactly the same value as with bool('yes') - which is True - and we definitely don't want that^^

I thought that we used to transform it to a boolean in l. 186 to 190:
```python
for key in list(set(_BOOL_KEYS) & set(adapted_form_data.keys())):
        if isinstance(adapted_form_data[key], str):
            adapted_form_data[key] = adapted_form_data[key] == 'yes'
        else:
            adapted_form_data[key] = bool(adapted_form_data[key])
```

Removing the key from the list `_BOOL_KEYS` in line 27 changes that.

Reply:
Ah I see. Puh..I think this is definitely a problem with the missing domain object/explicit "translation" layers..
Do you think I should change it back here and in Erica then? I have to admit that I completely missed that the code you posted takes YesNoFields into account.

Reply:
> Puh..I think this is definitely a problem with the missing domain object/explicit "translation" layers..

Yes :) I agree. I think that's why I was mentioning this so much when I was working at that part 😉 

I think it is not part of Erica. Therefore, I think it is not that important because we do not change another layer because of the form. I was just wondering what the advantage is of that change because I do not see one.